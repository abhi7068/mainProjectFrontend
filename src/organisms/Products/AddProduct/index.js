import React, { Component } from 'react';
import AddProductForm from '../../../molecules/AddProductForm';
import {
  Card,
  Select,
  MenuItem,
  CardHeader,
  FormControl,
  InputLabel,
} from '@material-ui/core';
import { withStyles } from '@material-ui/styles';
// import Button from "../../atoms/Button"
import { addProduct } from '../../../services/productService';
import { getCategories } from '../../../services/categoryServices';
import MaterialButton from '../../../atoms/MaterialButton';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import SnackBar from '../../../atoms/SnackBar';
import { connect } from 'react-redux';
const styles = {
  root: {
    width: '500px',
    height: '500px',
    margin: '10% auto',
  },
  label: {
    textTransform: 'capitalize',
  },
  display: {
    margin: '5% ',
  },
  formControl: {
    minWidth: 225,
  },
};

class index extends Component {
  constructor() {
    super();
    this.state = { data: {}, open: false, disabled: false };
    this.submitProduct = this.submitProduct.bind(this);
    this.onModelChange = this.onModelChange.bind(this);
  }

  submitProduct(event) {
    event.preventDefault();

    addProduct(this.state.data).then(
      this.setState({
        open: true,
        disabled: true,
      })
    );
  }

  onModelChange(event) {
    const name = event.target.name;

    const value = event.target.value;
    this.setState({ data: { ...this.state.data, [name]: value } });
  }

  componentDidMount() {
    this.props.dispatch(getCategories());
  }

  render() {
    return (
      <div>
        {this.state.open && (
          <SnackBar path="/admin/listproducts" name="Product Saved" />
        )}
        <Card
          classes={{
            root: this.props.classes.root,
          }}
        >
          <CardHeader title="ADD PRODUCT" />

          <form onSubmit={this.submitProduct}>
            <AddProductForm onModelChange={this.onModelChange} />
            {/*  Category:
          <Select name="category" onClick={this.onModelChange}>
            {this.state.categories.map((category, key) => (
              <MenuItem value={category._id} key={key}>
                {category.title}{' '}
              </MenuItem>
            ))}
          </Select> */}
            <FormControl className={this.props.classes.formControl}>
              <InputLabel>Category</InputLabel>
              <Select name="category" onClick={this.onModelChange}>
                {this.props.categories.data.map((category, key) => (
                  <MenuItem value={category._id} key={key}>
                    {category.title}{' '}
                  </MenuItem>
                ))}
              </Select>
            </FormControl>
            <br />
            <br />
            <MaterialButton
              data-testId="button"
              type="submit"
              disabled={this.state.disabled}
              name="Submit"
              // click={this.submitProduct}
            />
          </form>
        </Card>
      </div>
    );
  }
}
index.propTypes = {
  classes: PropTypes.object,
  root: PropTypes.object,
  display: PropTypes.object,
  history: PropTypes.object,
  dispatch: PropTypes.func,
  categories: PropTypes.array,
};
function mapStateToProps(state) {
  return {
    categories: state.categories,
  };
}

export default withRouter(connect(mapStateToProps)(withStyles(styles)(index)));
