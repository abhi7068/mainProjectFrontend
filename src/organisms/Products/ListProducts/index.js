import { withStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import EditIcon from '@material-ui/icons/Edit';
import RemoveCircleOutlineIcon from '@material-ui/icons/RemoveCircleOutline';
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import {
  getProducts,
  deleteProductById,
} from '../../../services/productService';
import { IconButton } from '@material-ui/core';

import SnackBar from '../../../atoms/SnackBar';
const StyledTableCell = withStyles((theme) => ({
  head: {
    backgroundColor: theme.palette.common.black,
    color: theme.palette.common.white,
  },
  body: {
    fontSize: 14,
  },
}))(TableCell);

const StyledTableRow = withStyles((theme) => ({
  root: {
    '&:nth-of-type(odd)': {
      backgroundColor: theme.palette.background.default,
    },
  },
}))(TableRow);
export default class index extends Component {
  constructor() {
    super();
    this.state = { products: [], open: false, disabled: false };
  }

  deleteProduct(id) {
    deleteProductById(id).then(
      this.setState({
        open: true,
        disabled: true,
      })
    );
  }

  componentDidMount() {
    getProducts().then((res) => {
      this.setState({ products: res.data });
    });
  }

  render() {
    return (
      <>
        {this.state.open && (
          <SnackBar path="/admin/dashboard" name="Product deleted" />
        )}
        <TableContainer /* style={{ marginLeft: '17.5% ', marginTop: '7%' }} */>
          <Table
            style={{ marginTop: '7%', marginLeft: '250px', width: '75.8%' }}
          >
            <TableHead>
              <TableRow>
                <StyledTableCell>Product Name</StyledTableCell>
                <StyledTableCell align="center">Description</StyledTableCell>
                <StyledTableCell align="center">Price</StyledTableCell>
                <StyledTableCell align="center">Discount</StyledTableCell>
                <StyledTableCell align="center">Category</StyledTableCell>
                <StyledTableCell align="center">Edit</StyledTableCell>
                <StyledTableCell align="center">Remove</StyledTableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {this.state.products.map((product, key) => (
                <StyledTableRow key={key}>
                  <StyledTableCell align="center">
                    {' '}
                    {product.product_name}
                  </StyledTableCell>

                  <StyledTableCell align="center">
                    {product.description}
                  </StyledTableCell>
                  <StyledTableCell align="center">
                    {product.price}
                  </StyledTableCell>
                  <StyledTableCell align="center">
                    {product.discount}
                  </StyledTableCell>
                  <StyledTableCell align="center">
                    {product.category ? product.category.title : ''}
                  </StyledTableCell>
                  <StyledTableCell align="center">
                    <Link to={`/admin/updateproduct/${product._id}`}>
                      <EditIcon />
                    </Link>
                  </StyledTableCell>
                  <StyledTableCell align="center">
                    <IconButton
                      button
                      disabled={this.state.disabled}
                      onClick={this.deleteProduct.bind(this, product._id)}
                    >
                      <RemoveCircleOutlineIcon name="delete" />
                    </IconButton>
                  </StyledTableCell>
                </StyledTableRow>
              ))}
            </TableBody>
          </Table>
        </TableContainer>
      </>
    );
  }
}
