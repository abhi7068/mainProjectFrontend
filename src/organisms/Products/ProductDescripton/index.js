import React, { Component } from 'react';
import { connect } from 'react-redux';
import Product from '../../../molecules/ProductDetails/ProductDetails';
import PropTypes from 'prop-types';
/* import axios from 'axios'; */
class index extends Component {
  render() {
    const {
      product: { products },
    } = this.props;
    //
    return (
      <div>
        <div>
          <Product details={products} />
        </div>
      </div>
    );
  }
}
const mapStateToProps = (state) => {
  return {
    product: state.product,
  };
};
index.propTypes = {
  product: PropTypes.object,
};
export default connect(mapStateToProps)(index);

/* class index extends Component {
  constructor(props) {
    super(props)
    this.state = {
       product:[]
    }
  }
  componentDidMount() {

  /*   axios.get() */
/*   }
  render() {

    return (
      <div>

      </div>
    )
  }
}
const mapStateToProps = (state) => {

  return {
    product: state

  };
};
index.propTypes = {
  product: PropTypes.object,
};
export default connect(mapStateToProps)(index);  */
