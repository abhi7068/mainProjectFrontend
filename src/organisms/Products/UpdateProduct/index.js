import React, { Component } from 'react';
import UpdateProductForm from '../../../molecules/UpdateProductForm';
import {
  Card,
  Select,
  MenuItem,
  FormControl,
  CardHeader,
  InputLabel,
} from '@material-ui/core';
import { withStyles } from '@material-ui/styles';
// import Button from "../../atoms/Button"
import {
  getProductsById,
  updateProductsById,
} from '../../../services/productService';
import { getCategories } from '../../../services/categoryServices';
import MaterialButton from '../../../atoms/MaterialButton';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import SnackBar from '../../../atoms/SnackBar';
import { connect } from 'react-redux';
const styles = {
  root: {
    width: '500px',
    height: '500px',
    margin: '10% auto',
  },
  label: {
    textTransform: 'capitalize',
  },
  display: {
    margin: '5% ',
  },
  formControl: {
    minWidth: 225,
  },
};

class index extends Component {
  constructor(props) {
    super(props);

    this.state = {
      data: {},
      open: false,
      sendUpdateId: this.props.id,
      disabled: false,
    };
    this.updateProduct = this.updateProduct.bind(this);
    this.onModelChange = this.onModelChange.bind(this);
  }

  updateProduct(event) {
    event.preventDefault();
    updateProductsById(this.state.sendUpdateId, this.state.data).then(
      this.setState({
        open: true,
        disabled: true,
      })
    );
  }

  componentDidUpdate(prevProps, prevState) {
    //  const { dispatch } = this.props;
    if (prevState.sendUpdateId !== this.state.sendUpdateId) {
      getProductsById(this.state.sendUpdateId).then((res) => {
        this.setState({ data: res.data });
      });
    }
  }

  static getDerivedStateFromProps(props, state) {
    //
    if (props.id !== state.sendUpdateId) {
      return { sendUpdateId: this.props.id };
    }
    return null;
  }

  onModelChange(event) {
    const name = event.target.name;
    const value = event.target.value;
    this.setState({ data: { ...this.state.data, [name]: value } });
  }

  componentDidMount() {
    this.props.dispatch(getCategories());
    getProductsById(this.state.sendUpdateId).then((res) => {
      this.setState({ data: res.data });
    });
  }

  render() {
    const {
      categories: { data },
    } = this.props;

    //
    return (
      <div>
        {this.state.open && (
          <SnackBar path="/admin/listproducts" name="Product Saved" />
        )}

        <Card
          classes={{
            root: this.props.classes.root,
          }}
        >
          <CardHeader title="UPDATE PRODUCT" />
          <form onSubmit={this.updateProduct}>
            <UpdateProductForm
              onModelChange={this.onModelChange}
              value={this.state.data}
            />

            <FormControl className={this.props.classes.formControl}>
              <InputLabel id="demo-simple-select-label">Category</InputLabel>
              <Select name="category" onClick={this.onModelChange}>
                {data.map((category, key) => (
                  <MenuItem value={category._id} key={key}>
                    {category.title}{' '}
                  </MenuItem>
                ))}
              </Select>
            </FormControl>
            <br />
            <br />
            <MaterialButton
              data-testId="button"
              disabled={this.state.disabled}
              name="Submit"
              type="submit"
            />
          </form>
        </Card>
      </div>
    );
  }
}
index.propTypes = {
  classes: PropTypes.object,
  root: PropTypes.object,
  display: PropTypes.object,
  id: PropTypes.string,
  history: PropTypes.object,
  dispatch: PropTypes.func,
  categories: PropTypes.array,
};
function mapStateToProps(state) {
  return {
    categories: state.categories,
  };
}
export default withRouter(connect(mapStateToProps)(withStyles(styles)(index)));
