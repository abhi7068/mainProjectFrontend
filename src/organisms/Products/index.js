import React, { Component } from 'react';
import { connect } from 'react-redux';
import Product from '../../molecules/Product';
import {
  getProducts,
  updateProducts,
  resetProducts,
} from '../../store/actions/productAction';
import { getProductByCategory } from '../../services/productService';
import CircularProgress from '@material-ui/core/CircularProgress';
import classes from './product.module.css';
import PropTypes from 'prop-types';
class Products extends Component {
  componentDidMount() {
    const {
      categoryId,
      updateProducts,
      getProducts,
      url,
      resetProducts,
    } = this.props;

    resetProducts();
    if (categoryId && url !== '/') {
      getProductByCategory(categoryId).then((result) => {
        updateProducts(result.data);
      });
    } else getProducts();
  }

  render() {
    const { isLoading, title, productList } = this.props;

    return (
      <div style={{ marginBottom: '4em' }}>
        {isLoading ? (
          <CircularProgress disableShrink />
        ) : (
          <>
            <h2>{title || ''}</h2>
            <div className={classes.container}>
              {productList.length ? (
                productList.map((product, i) => {
                  return (
                    <div key={product._id} className={classes.boxes}>
                      <Product product={product} />
                    </div>
                  );
                })
              ) : (
                <h2>No Products</h2>
              )}
            </div>
          </>
        )}
      </div>
    );
  }
}
const mapStateToProps = (state) => {
  return {
    productList: state.product.filteredLatest,
    isLoading: state.product.isLoading,
    activeCategoryId: state.product.activeCategoryId,
  };
};
Products.propTypes = {
  productList: PropTypes.array,
  dispatch: PropTypes.func,
  isLoading: PropTypes.bool,
  categoryId: PropTypes.string,
  updateProducts: PropTypes.string,
  activeCategoryId: PropTypes.string,
  getProducts: PropTypes.func,
  resetProducts: PropTypes.func,
  url: PropTypes.string,
  title: PropTypes.string,
};
export default connect(mapStateToProps, {
  updateProducts,
  resetProducts,
  getProducts,
})(Products);
