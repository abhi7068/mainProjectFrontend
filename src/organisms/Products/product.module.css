.container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
}
.boxes {
  padding: 4vmin 4vmin;
}
