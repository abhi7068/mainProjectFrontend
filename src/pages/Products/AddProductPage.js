import React, { Component } from 'react';
import AddProduct from '../../organisms/Products/AddProduct';
export default class AddProductPage extends Component {
  render() {
    return (
      <div>
        <AddProduct />
      </div>
    );
  }
}
