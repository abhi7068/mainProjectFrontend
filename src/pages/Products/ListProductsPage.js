import React, { Component } from 'react';
import ListProducts from '../../organisms/Products/ListProducts';

export default class ListProductsPage extends Component {
  render() {
    return (
      <div>
        <ListProducts />
      </div>
    );
  }
}
