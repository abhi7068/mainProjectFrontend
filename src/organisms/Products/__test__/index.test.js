import React from 'react';
import Products from '../index';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import reducer from '../../../store/reducers/rootReducer';
import thunk from 'redux-thunk';
/* import axiosMock from 'axios'; */
import { BrowserRouter } from 'react-router-dom';
const store = createStore(reducer, applyMiddleware(thunk));
afterEach(cleanup);

it('matches snapshot', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Products product="" />
      </BrowserRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
