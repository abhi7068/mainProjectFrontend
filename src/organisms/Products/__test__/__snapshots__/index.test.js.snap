// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="margin-bottom: 4em;"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleDisableShrink MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
